example/socks/socks_client.hpp

```diff
@@ -0,0 +1,491 @@
+﻿//
+// Copyright (c) 2018 jackarain (jack dot wgm at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#pragma once
+
+#include "error.hpp"
+
+#include <boost/asio/spawn.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/streambuf.hpp>
+#include <boost/asio/read.hpp>
+#include <boost/asio/write.hpp>
+
+#include <iostream>
+#include <string>
+#include <memory>
+
+using boost::asio::ip::tcp;
```

Comment:
We shouldn't have a `using` directive at file scope in a header file. I know it sucks typing all of that out but it is better for users if we don't do this.